### PR TITLE
feat: Add detailed assembly reporting to enable Vulnerability Management support.

### PR DESCRIPTION
**Pull request by jaffinito**
## Description

- Adds support for Vulnerability Management via a new service, UpdateLoadedModulesService and supported types.  Sends new assembly data on a one minute cycle.  Only new assemblies are reported so it is expected that nothing will be sent from longer running apps.
- Adds unit tests for new code.
- Adds configuration to the UpdateLoadedModulesService so that we can change the cycle time for testing.
- Adds an integration test for Vulnerability Management to AgentFeatures group for Framework and Core.  Only one test was needed since the unit test coverage is high.

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

File: src/Agent/NewRelic/Agent/Core/JsonConverters/LoadedModuleWireModelCollectionJsonConverter.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+using System;
+using NewRelic.Agent.Core.WireModels;
+using Newtonsoft.Json;
+
+namespace NewRelic.Agent.Core.JsonConverters
+{
+    public class LoadedModuleWireModelCollectionJsonConverter : JsonConverter<LoadedModuleWireModelCollection>
+    {
+        private const string JarsName = "Jars";
```
Comment (chynesNR) [inline src/Agent/NewRelic/Agent/Core/JsonConverters/LoadedModuleWireModelCollectionJsonConverter.cs:12]: I'm not sure where the best place is for this, but we might want to document why we're sending a list of "jars" in .NET. :)

Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/JsonConverters/LoadedModuleWireModelCollectionJsonConverter.cs:12]: Yeah, that is a quirk of the system.  It was meant for Java.  Once the VM team gets some time with the data we send, they will loop back around and let us know what is required and we can trim down what we send.  At the same time we can push for the team to update the endpoint to have an "Assemblies" or similar label for us,

I'll add a note.

File: src/Agent/NewRelic/Agent/Core/Utilities/UpdatedLoadedModulesService.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        {
+            _configurationService = configurationService;
+            _dataTransportService = dataTransportService;
+            _scheduler = scheduler;
+            _scheduler.ExecuteEvery(GetLoadedModules, _configuration.UpdateLoadedModulesCycle);
+        }
+
+        private void GetLoadedModules()
+        {
+            var assemblies = AppDomain.CurrentDomain.GetAssemblies()
```
Comment (chynesNR) [inline src/Agent/NewRelic/Agent/Core/Utilities/UpdatedLoadedModulesService.cs:32]: I'm still a little concerned about performance if a large number of assemblies are loaded, but I think it's ok just to keep an eye on it for now.

**Review by chynesNR — APPROVED** [review]

**Review by tippmar-nr — APPROVED** [review]

File: src/Agent/NewRelic/Agent/Core/Utilities/UpdatedLoadedModulesService.cs (hunk 2 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/Utilities/UpdatedLoadedModulesService.cs:32]: If I pulled in the AgentHealthReporter I could make a few supportability metrics.  Add them to Angler so we could track them.

File: src/Agent/NewRelic/Agent/Core/JsonConverters/LoadedModuleWireModelCollectionJsonConverter.cs (hunk 1 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/JsonConverters/LoadedModuleWireModelCollectionJsonConverter.cs:13]: Updated.

File: src/Agent/NewRelic/Agent/Core/Utilities/UpdatedLoadedModulesService.cs (hunk 2 above)
Comment (nrcventura) [reply src/Agent/NewRelic/Agent/Core/Utilities/UpdatedLoadedModulesService.cs:32]: I think that we already have timing code in the agent that we could leverage for this. We used them in some infinite tracing code.

Comment (nrcventura) [reply src/Agent/NewRelic/Agent/Core/Utilities/UpdatedLoadedModulesService.cs:32]: The AgentTimerService is what I'm thinking of.

**Pull request merged**